configargparse.py

```diff
@@ -244,9 +255,25 @@ def serialize(self, items, default_flow_style=False):
         # lazy-import so there's no dependency on yaml unless this class is used
         yaml = self._load_yaml()
 
+        def _set_lvlkey(d, k, v):
+            # set nested dict key; e.g
+            # my-config-opt -> { 'my': { 'config': { 'opt': v } } }
+            parts = k.split('-')
```

Comment:
doesn't this change behavior for regular key-value pairs where the key has `-` in it?

eg.  my-key: 3